docker/jupyter/custom/jupyter_delocalize.py

```diff
@@ -239,8 +256,13 @@ def save(self, model, path=''):
     return ret
 
   def rename_file(self, old_path, new_path):
-    # TODO(IA-1028): Integrate Welder renaming.
-    super(WelderContentsManager, self).rename_file(old_path, new_path)
+    if not self._check_welder_edit_mode(old_path) and not self._check_welder_edit_mode(new_path):
+      # If we're not touching any edit mode files, just do a normal move.
+      return super(WelderContentsManager, self).rename_file(old_path, new_path)
+
+    # These methods already properly handle edit mode semantics.
+    self.save(self.get(old_path), new_path)
+    self.delete(old_path)
```

Comment:
I think we'll need to revert the save if `delete` fails

Reply:
Good catch. These two-phase actions make me a bit nervous - but we'll need to shift this logic into Welder to fix it.

Reply:
(added)

Reply:
I feel like a rename api in welder will be tricky to implement as well. How about welder provides /checkLock api?

Reply:
`/checkLock` would not help here.

Actually on further research of the GCS APIs - looks like move is not a real operation: https://cloud.google.com/storage/docs/renaming-copying-moving-objects#storage-move-object-java  so there is not as much of an advantage as I'd thought to moving this into Welder - it would still need to do a two-phase approach.